storage/core/src/main/java/com/hortonworks/streamline/storage/impl/jdbc/JdbcStorageManager.java

```diff
@@ -61,14 +63,17 @@ public JdbcStorageManager(QueryExecutor queryExecutor) {
     @Override
     public void add(Storable storable) throws AlreadyExistsException {
         log.debug("Adding storable [{}]", storable);
-        final Storable existing = get(storable.getStorableKey());
-
-        if(existing == null) {
-            addOrUpdate(storable);
-        } else if (!existing.equals(storable)) {
-            throw new AlreadyExistsException("Another instance with same id = " + storable.getPrimaryKey()
-                    + " exists with different value in namespace " + storable.getNameSpace()
-                    + ". Consider using addOrUpdate method if you always want to overwrite.");
+
+        try {
+            queryExecutor.insert(storable);
+        } catch (StorageException ex) {
+            if (ex.getCause() instanceof SQLIntegrityConstraintViolationException) {
+                throw new AlreadyExistsException("Another instance with same id = " + storable.getPrimaryKey()
```

Comment:
@HeartSaVioR That logic looks convoluted. Are you suggesting the below?
```
try {
            queryExecutor.insert(storable);
        } catch (StorageException ex) {
            Throwable cause = ex.getCause();
            if (cause instanceof SQLIntegrityConstraintViolationException) {
                // below check is the suggestion!!
                if(get(storable.getStorableKey()) == null) {
                    throw new AlreadyExistsException("Another instance with same id = " + storable.getPrimaryKey()
                                                             + " exists with different value in namespace " + storable.getNameSpace()
                                                             + ". Consider using addOrUpdate method if you always want to overwrite.");
                }
            }

            throw ex;
        }
```
>If "select query" fails, we could treat it as AlreadyExistsException.
This logic looks confusing as `AlreadyExistsException` is thrown when there is no entry for the given Storable.